engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -860,6 +860,11 @@ public void saveExtraDhcpOptions(final String networkUuid, final Long nicId, fin
 
         NicVO vo = new NicVO(guru.getName(), vm.getId(), network.getId(), vm.getType());
 
+        DataCenterVO dcVo = _dcDao.findById(network.getDataCenterId());
+        if (dcVo.getNetworkType() == NetworkType.Basic) {
+            configureNicProfileBasedOnRequestedIp(requested, profile, network);
```

Comment:
@fmaximus I changed the execution flow; now this workflow will be executed just in case of a basic network.

After this change I could add IPv4 either in an advanced network, with a 'quest IP', and in a basic network with a 'public IP' (from the IPAddressVO in this case). Thanks again for pointing that issue.